### PR TITLE
Two new data fixes

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping [hunk 1]
```diff
@@ -0,0 +1,20 @@
+CLASS net/minecraft/class_5946 net/minecraft/datafixer/fix/ArrowPickupFix
+	COMMENT A fix that automatically converts the {@code player} NBT field in arrows
+	COMMENT to {@code pickup} field if possible.
```
Comment (haykam821) [inline mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping]:
```suggestion
	COMMENT A fix that automatically renames the {@code player} byte in arrow data to
	COMMENT {@code pickup}, if there is not any existing {@code pickup} data.
```

**Review by haykam821 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping]: Sure, though I don't see the difference

Comment (haykam821) [reply mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping]: I think yarn should avoid referring to the contents of NBT compounds as fields

File: mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+		ARG 1 outputSchema
+	METHOD method_34684 update (Lcom/mojang/datafixers/Typed;)Lcom/mojang/datafixers/Typed;
+		ARG 1 typed
+	METHOD method_34685 updateEntity (Lcom/mojang/datafixers/Typed;Ljava/lang/String;Ljava/util/function/Function;)Lcom/mojang/datafixers/Typed;
+		ARG 1 typed
+		ARG 2 choiceName
+		ARG 3 updater
+	METHOD method_34686 update (Lcom/mojang/serialization/Dynamic;)Lcom/mojang/serialization/Dynamic;
+		COMMENT When the {@code pickup} NBT field of an arrow's data is absent, sets it
+		COMMENT from the arrow's {@code player} NBT field.
```
Comment (liach) [inline mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping]:
```suggestion
		COMMENT When the {@code pickup} NBT byte of an arrow's data is absent, sets it
		COMMENT from the arrow's {@code player} NBT byte.
```

File: mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/datafixer/fix/ArrowPickupFix.mapping]: Ah. Now I've fixed the other reference of NBT compound element as "field" in javadoc, mind review again?

**Comment by liach** [comment]
Requesting reviews. Shoutout to discord peep.

**Review by haykam821 — APPROVED** [review]

**Pull request merged**